mllib/src/test/scala/org/apache/spark/ml/classification/GBTClassifierSuite.scala

```diff
@@ -365,6 +365,20 @@ class GBTClassifierSuite extends MLTest with DefaultReadWriteTest {
     assert(mostImportantFeature !== mostIF)
   }
 
+  test("model evaluateEachIteration") {
+    for (lossType <- Seq("logistic")) {
+      val gbt = new GBTClassifier()
+        .setMaxDepth(2)
+        .setMaxIter(2)
+        .setLossType(lossType)
+      val model = gbt.fit(trainData.toDF)
+      val eval1 = model.evaluateEachIteration(validationData.toDF)
+      val eval2 = GradientBoostedTrees.evaluateEachIteration(validationData,
```

Comment:
This is testing the spark.ml implementation against itself.  I was about to recommend using the old spark.mllib implementation as a reference.   However, the old implementation is not tested at all.  Would you be able to test against a standard implementation in R or scikit-learn (following the patterns used elsewhere in MLlib)?

Reply:
I search scikit-learn doc, there seems no similar method like `evaluateEachIteration`, we can only use `staged_predict` in `sklearn.ensemble.GradientBoostingRegressor` and then implement almost the whole logic again. In R package I also do not find this method.
Now I update the unit test, to compare with hardcoded result.